lib/fhir_models/bootstrap/definitions.rb

```diff
@@ -216,21 +216,27 @@ def self.get_codes(uri)
     def self.get_display(uri, code)
       return nil if uri.nil? || code.nil?
       load_expansions
-      valuesets = @@expansions.select { |x| x['compose']['include'].detect {|i| i['system'] == uri } }
-      valuesets += @@valuesets.select { |x| x['url'] == uri }
+      valuesets = @@expansions.select { |ex| ex['compose']['include'].detect { |i| i['system'] == uri } }
+      valuesets += @@valuesets.select { |vs| vs['url'] == uri }
       code_hash = nil
       valuesets.each do |valueset|
-        if !valueset['expansion'].nil? && !valueset['expansion']['contains'].nil?
-          code_hash = valueset['expansion']['contains'].select { |x| x['system'] == uri && x['code'] == code}.first
-        elsif !valueset['compose'].nil? && !valueset['compose']['include'].nil?
-          included_systems = valueset['compose']['include'].map { |x| x['system'] }.uniq
-          systems = @@valuesets.select { |x| x['resourceType'] == 'CodeSystem' && included_systems.include?(x['url']) }
-          systems.each do |x|
-            code_hash = x['concept'].select { |con| con['url'] == uri && con['code'] == code }.first if x['concept']
+        if valueset['expansion'] && valueset['expansion']['contains']
+          # This currently only matches 'expansions', not 'valuesets'
+          code_hash = valueset['expansion']['contains'].detect { |contained| contained['system'] == uri && contained['code'] == code }
+        elsif valueset['compose'] && valueset['compose']['include']
+          # I'm not sure this branch ever matches, see comment below.
+          included_systems = valueset['compose']['include'].map { |inc| inc['system'] }.uniq
+          code_systems = @@valuesets.select { |vs| vs['resourceType'] == 'CodeSystem' && included_systems.include?(vs['url']) }
+          code_systems.each do |code_system|
+            # CodeSystem.concept doesn't have a 'url' attribute, so it should
+            # never match on this line.
+            # See http://build.fhir.org/codesystem.html
+            code_hash = code_system['concept'].detect { |con| con['url'] == uri && con['code'] == code } if code_system['concept']
             break if code_hash
           end
         elsif valueset['concept']
-          code_hash = valueset['concept'].select { |v| v['code'] == code }.first
+          # This currently only matches 'valuesets', not 'expansions'
```

Comment:
Yes, my proposal is that we might only run this logical branch against `valuesets` — but if there's a good possibility of that being insufficient "soon," then that's okay too! (Just thinking that if we're running against hundreds or more ValueSets that we don't need to ... we might save a little effort.) :)